src/main/scala/performanceanalysis/server/Protocol.scala

```diff
@@ -122,7 +132,23 @@ object Protocol {
 }
 
 trait Protocol extends DefaultJsonProtocol {
-  implicit val metricFormatter = jsonFormat(Metric.apply, "metric-key", "regex")
+
+  implicit object valueTypeFormat extends JsonFormat[ValueType] {
+    def read(value: JsValue): ValueType = value match {
+      case JsString("string") => ValueType(classOf[String])
+      case JsString("boolean") => ValueType(classOf[Boolean])
+      case JsString("duration") => ValueType(classOf[Duration])
+      case _ => deserializationError("Unknown value type")
+    }
+    def write(f: ValueType): JsValue = f.aType match {
+      case c if c == classOf[String] => JsString("string")
```

Comment:
not sure if it will work, here I'm expecting to match the Class object (String.class) and not an instance of that Class (String)